composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -217,33 +210,75 @@ def apply(self, event: Event, state: State, logger: Logger) -> Optional[int]:
                     'Sequence Length Warmup requires the `state.dataloader` to have a `batch_size` attribute.') from e
             if per_gpu_macrobatch is None:
                 raise RuntimeError('Sequence Length Warmup algorithm requires constant batch size.')
-            per_gpu_batch = ceil(per_gpu_macrobatch / state.grad_accum)
-
-            input_ids = torch.randint(low=0,
-                                      high=vocab_size - 1,
-                                      size=(per_gpu_batch, self.max_seq_length),
-                                      device=device).long()
-            labels = input_ids.clone()
-            attn_mask = torch.ones_like(labels)
-            model_inputs = {
-                'input_ids': input_ids,
-                'labels': labels,
-                'attention_mask': attn_mask,
-            }
-
-            # start by running a forward and backward pass
-            # of the maximum sequence length to allocate cache.
-            with get_precision_context(state.precision):
-                outputs = state.model.forward(model_inputs)
-                loss = self._original_model.loss(outputs, model_inputs)
-
-            # since use_grad_scaling is in the Trainer, and we
-            # don't care about the loss values, skip scaling
-            for loss_item in ensure_tuple(loss):
-                loss_item.backward()
-
-            for optimizer in state.optimizers:
-                optimizer.zero_grad()
+
+            batch_clone = {}
+            device_batch_size = 0
+            device = None
+            for k, v in state.batch.items():
```

Comment:
is there a reason to use the `state.batch` directly, versus creating a random tensor as before?

Reply:
It seemed more straightforward to use the `state.batch` since it reflects the structure of actual training batches. The alternative required inferring (or hardcoding) the batch structure in a way that seemed less flexible and/or harder to maintain if models change.
If I'm missing an upside to using random tensors, though, please let me know.

Reply:
OK that makes sense. One style nit is to break this up into a few logical bits, e.g. clone everything first, then truncate.
```
batch_clone = {k: torch.clone(v) for k, v in state.batch.items()]

# truncate all sequence-shaped tensors to the max sequence length
for k, v in batch_clone.items():
    if len(v.shape) > 1:
        batch_clone[v] = batch_clone[v][:, :self.max_seq_length]

device_batch_size = 
device = 
```

It's less efficient, but more readable

Reply:
Changes applied